99_helpers/create_sample_data.R

```diff
@@ -0,0 +1,355 @@
+fetch_random_pin_data <-
+  function(admin_level = 2,
+           gender_age_disagg = FALSE,
+           pop_group_disagg = FALSE) {
+
+    # read sectoral PiN
+    df <- read_csv(
+      file.path(
+        file_paths$output_dir,
+        "datasets",
+        "2022_hno_pin_cluster_totals.csv"
+      )
+    ) %>%
+      mutate(
+        adm_name = case_when(
+          adm0_pcode == "COL" ~ tolower(adm1_name),
+          !is.na(adm3_name) ~ tolower(adm3_name),
+          !is.na(adm2_name) ~ tolower(adm2_name),
+          TRUE ~ tolower(adm1_name)
+        ),
+        adm_name = gsub("[ ]|[-]|[.]|[_]|[,]", "", adm_name),
+        adm_pcode = case_when(
+          admin_level < 2 ~ adm1_pcode,
+          admin_level > 2 ~ adm3_pcode,
+          TRUE ~ adm2_pcode
+        ),
+        age = case_when(
+          is.na(age) ~ NA_character_,
+          age %in% c("< 18 ans", "child", "below_18", "children") ~ "Children",
+          TRUE ~ "Adult",
+        ),
+        sex = case_when(
+          is.na(sex) ~ NA_character_,
+          sex %in% c("female", "Femme", "f") ~ "Female",
+          sex %in% c("male", "Homme", "m") ~ "Male"
+        ),
+        population_group = case_when(
+          population_group %in% c(
+            "idp",
+            "idps",
+            "IDPs",
+            "In-Camp-IDPs",
+            "Out-of-Camp-IDPs",
+            "displaced",
+            "pdi",
+            "PDI"
+          ) ~ "IDPs",
+          population_group %in% c(
+            "APV",
+            "Autres populations",
+            "vulnerable_people",
+            "vul",
+            "other"
+          ) ~ "Other-vulnerable-people",
+          population_group == "shock_affected" ~ "Shock-affected",
+          population_group %in% c(
+            "refugees",
+            "ref_car",
+            "ref_nga",
+            "ref_other",
+            "ref",
+            "Refugees"
+          ) ~ "Refugees",
+          population_group %in% c(
+            "host",
+            "Non-displaced",
+            "non_displaced",
+            "non_pdi",
+            "residents",
+            "Residents"
+          ) ~ "Residents",
+          population_group %in% c(
+            "ret",
+            "Retournes",
+            "return",
+            "Rapatries",
+            "returnees",
+            "Returnees"
+          ) ~ "Returnees"
+        )
+      ) %>%
+      arrange(
+        adm0_pcode,
+        adm1_pcode,
+        adm2_pcode,
+        adm3_pcode,
+        sector,
+        population_group,
+        sex,
+        desc(age)
+      )
+
+    # read MSNA dataset and prepare it for alignement with sectoral PiN
+    df_msna <- read_csv(file.path(
+      dirname(getwd()),
+      "MSNA_data.csv"
```

Comment:
Since we are okay to share the MSNA data in the JIAF folder, can you move into there and reconfigure all of the our code to read the MSNA data straight from the Sharepoint?